pkg/secretpath/path.go

```diff
@@ -0,0 +1,226 @@
+// Package secretpath implements utility functions for manipulating paths
+// compatible with SecretHub, e.g. namespaces, repositories, directories,
+// secrets and versions.
+package secretpath
+
+import (
+	"strconv"
+	"strings"
+
+	"bitbucket.org/zombiezen/cardcpx/natsort"
+)
+
+const (
+	elemSepByte    = '/'
+	elemSep        = string(elemSepByte)
+	versionSepByte = ':'
+	versionSep     = string(versionSepByte)
+	latestSuffix   = ":latest"
+)
+
+// Join joins any number of elements into a path, adding a separator
+// if necessary. Empty string elements are ignored.
+func Join(elements ...string) string {
+	result := ""
+	for _, e := range elements {
+		e = strings.Trim(e, elemSep)
+		if len(e) != 0 && len(result) == 0 {
+			result += e
+		} else if len(e) != 0 {
+			result += elemSep + e
+		}
+	}
+
+	return Clean(result)
+}
+
+// HasVersion returns true when a version suffix is specified in the path.
+func HasVersion(path string) bool {
+	return hasVersion(path)
+}
+
+// hasVersion returns true when a version suffix is specified in the path.
+func hasVersion(path string) bool {
```

Comment:
Why have this private func if there's a public function that does exactly the same: `HasVersion`?

Reply:
Ah, these used to be different. Now there not. You're right, the private function can be removed, moving the code to the public function.